File: examples/node/asticaDesign_sample.js
const axios = require('axios');

var asticaAPI_timeout = 120;
var asticaAPI_key = 'YOUR API KEY'; // Put your API key here

// See Input Documentation:  https://astica.ai/design/documentation/#inputs

var asticaAPI_endpoint = 'https://design.astica.ai/generate_image';
var asticaAPI_modelVersion = '2.0_full';
var asticaAPI_prompt = 'close-up photography of older gentleman standing in the rain at night, in a street lit by lamps'; 
var asticaAPI_prompt_negative = '';
var asticaAPI_moderate = 1; //1 = Moderate Generation; 0 = Allow NSFW
var asticaAPI_generate_quality = 'faster';//high, standard, fast, faster
var asticaAPI_generate_lossless = 0; //0 = Default JPG, 1 = lossless uncompressed PNG
var asticaAPI_seed = 0; //0 will randomize the seed for every generation
var asticaAPI_low_priority = 0;


//Prepare payload
var asticaAPI_payload = {
    tkn: asticaAPI_key,
    modelVersion: asticaAPI_modelVersion,
    prompt: asticaAPI_prompt,
    prompt_negative: asticaAPI_prompt_negative,
    moderate: asticaAPI_moderate,
    generate_quality: asticaAPI_generate_quality,
    generate_lossless: asticaAPI_generate_lossless,
    seed: asticaAPI_seed,
    low_priority: asticaAPI_low_priority,
};

//Demo function
(async () => {    
    // Submit request    
    console.log("Generating Image..")
    const result = await asticaAPI(asticaAPI_endpoint, asticaAPI_payload, asticaAPI_timeout);    
    console.log('astica API Output:\n', JSON.stringify(result, null, 4));
    // Handle response
    if (result.status === 'error') {
        console.log('Error:', result.error);
    } else if (result.status === 'success') {
        if (result.resultURI) {
            console.log('Output URI:', result.resultURI, '\nQuery this URL to obtain the output of your results');
        } else {
            console.log('===============');
            console.log('Generated Image:', result.output);
            console.log('===============');
        }
    } else { console.log('Invalid response'); }
    
    
})();


// Required astica api function
async function asticaAPI(endpoint, payload, timeout) {
    try {
        const response = await axios.post(endpoint, payload, {
            headers: { 'Content-Type': 'application/json' },
            timeout: timeout * 1000, // timeout in ms
        });

        if (response.status === 200) {
            return response.data;
        }
    } catch (error) {
        return { status: 'error', error: 'Failed to connect to the API.' };
    }
}

File: examples/python/asticaDesign_sample.py
import requests
import json

# API configurations
asticaAPI_key = 'YOUR API KEY'  # visit https://astica.ai
asticaAPI_timeout = 60 # in seconds.

#See Input Documentation:  https://astica.ai/design/documentation/#inputs

asticaAPI_endpoint = 'https://design.astica.ai/generate_image'
asticaAPI_modelVersion = '2.0_full'
asticaAPI_prompt = 'close-up photography of older gentleman standing in the rain at night, in a street lit by lamps'
asticaAPI_prompt_negative = ''
asticaAPI_generate_quality = 'faster' #high, standard, fast, faster
asticaAPI_generate_lossless = 0 #0 = Default JPG, 1 = lossless uncompressed PNG
asticaAPI_seed = 0 #0 will randomize the seed for every generation
asticaAPI_moderate = 1
asticaAPI_low_priority = 0 #0 = realtime, 1 = low_priority (lower cost)

# Prepare payload
asticaAPI_payload = {
    'tkn': asticaAPI_key,
    'modelVersion': asticaAPI_modelVersion,
    'prompt': asticaAPI_prompt,
    'prompt_negative': asticaAPI_prompt_negative,
    'generate_quality': asticaAPI_generate_quality,
    'generate_lossless': asticaAPI_generate_lossless,
    'seed': asticaAPI_seed,
    'moderate': asticaAPI_moderate,
    'low_priority': asticaAPI_low_priority,
}


def asticaAPI(endpoint, payload, timeout):
    response = requests.post(endpoint, data=json.dumps(payload), timeout=timeout, headers={ 'Content-Type': 'application/json', })
    if response.status_code == 200:
        return response.json()
    else:
        return {'status': 'error', 'error': 'Failed to connect to the API.'}



# call API function and store result
asticaAPI_result = asticaAPI(asticaAPI_endpoint, asticaAPI_payload, asticaAPI_timeout)

# print API output
# Handle asticaAPI response
if 'status' in asticaAPI_result:
    # Output Error if exists
    if asticaAPI_result['status'] == 'error':
        print('Error:', asticaAPI_result['error'])
    # Output Success if exists
    elif asticaAPI_result['status'] == 'success':
        if 'resultURI' in asticaAPI_result:
            print('===============')
            print('Low Priority URI: ', asticaAPI_result['resultURI'], '\nQuery this URL to obtain the output of your results')
            print('===============')
        else:
            print('===============')
            print('Generated Image:', asticaAPI_result['output'])
            print('===============')
    print('\nastica API Output:')
    print(json.dumps(asticaAPI_result, indent=4))
    print('=================')
           
else:
    print('Invalid response')